boilerplate/openshift/golang-osd-operator/csv-generate/csv-generate.sh

```diff
@@ -0,0 +1,148 @@
+#!/bin/bash -x
+
+usage() { echo "Usage: $0 -o operator_name -i operator_image -c saas-repository-channel [-t]" 1>&2; exit 1; }
+
+# TODO : Add support of long-options 
+while getopts "c:dg:h:i:n:o:" option; do
+    case "${option}" in
+        c)
+            operator_channel=${OPTARG}
+            ;;
+        d)
+            diff_generate=true
+            ;;
+        g)
+            generate_script=${OPTARG}
+            ;;
+        h)
+            operator_commit_hash=${OPTARG}
+            ;;
+        i)
+            operator_image=${OPTARG}
+            ;;
+        n)
+            operator_commit_number=${OPTARG}
+            ;;
+        o)
+            operator_name=${OPTARG}
+            ;;
+        *)
+            usage
+    esac
+done
+
+# Checking parameters
+unset csv_missing_param_error
+
+if [ -z "$operator_channel" ] ; then 
+    echo "Missing operator_channel parameter"
+    csv_missing_param_error=true
+fi
+
+if [ -z "$operator_image" ] ; then 
+    echo "Missing operator_image parameter"
+    csv_missing_param_error=true
+fi
+
+if [ -z "$operator_name" ] ; then 
+    echo "Missing operator_name parameter"
+    csv_missing_param_error=true
+fi
+
+if [ -z "$operator_commit_hash" ] ; then 
+    echo "Missing operator_commit_hash parameter"
+    csv_missing_param_error=true
+fi
+
+if [ -z "$operator_commit_number" ] ; then 
+    echo "Missing operator_commit_number parameter"
+    csv_missing_param_error=true
+fi
+
+if [ -z "$generate_script" ] ; then 
+    echo "Missing generate_script parameter"
+    csv_missing_param_error=true
+fi
+
+if [ ! -z "$csv_missing_param_error" ] ; then
+    usage
+fi
+
+# If no override, using the gitlab repo
+if [ -z "$GIT_PATH" ] ; then 
+    GIT_PATH="https://app:@gitlab.cee.redhat.com/service/saas-${operator_name}-bundle.git"
+fi
+
+# Calculate previous version
+SAAS_OPERATOR_DIR="saas-${operator_name}-bundle"
+BUNDLE_DIR="$SAAS_OPERATOR_DIR/${operator_name}/"
+
+if [ "$diff_generate" = true ] ; then
+    OPERATOR_NEW_VERSION=$(ls "$BUNDLE_DIR" | sort -t . -k 3 -g | tail -n 1)
```

Comment:
@bdematte doesn't look like this is addressed unless there is something I'm misunderstanding? Since you `rm` the  `$SAAS_OPERATOR_DIR` in the next else branch I guess you expect it to exist if `$diff_generate = true` ? We should validate that before this if statement as this would fail and the `rm` in the next branch would exit because `set -e` is set.